### PR TITLE
Server can choose the media type on Accept fail

### PR DESCRIPTION
**Pull request by benjie**
If the client issues `Accept: application/graphql-response+json, application/json` and the server does not support this for the request (e.g. the request uses `@stream`/`@defer` so needs a multipart response) then the server should be able to choose to respond with another suitable media type, for example `multipart/mixed;type=application/graphql-response+json`. Thanks to @michaelstaib for pointing out this oversight.

(Note: v1 doesn't explicitly specify `@stream`/`@defer` or `subscription` related concerns, but we should still be future proof where possible.)

### CODE REVIEW

**Review by michaelstaib — APPROVED** [review]

**Review by Shane32 — APPROVED** [review]

**Review by enisdenjo — APPROVED** [review]

File: spec/GraphQLOverHTTP.md [hunk 1]
```diff
@@ -426,8 +426,8 @@ In alignment with the
 specification, when a client does not include at least one supported media type
 in the `Accept` HTTP header, the server MUST either:
 
-1. Disregard the `Accept` header and respond with the default media type of
-   `application/json`, specifying this in the `Content-Type` header; OR
+1. Disregard the `Accept` header and respond with the server's choice of media
+   type, indicating this in the `Content-Type` header; OR
```
Comment (ghmcadams) [inline spec/GraphQLOverHTTP.md:430]: Would a suggestion make sense here? (if you're not sure what to choose, use `application/json`)

**Comment by maraisr** [comment]
I thought we want the exact opposite? ~> https://github.com/robrichard/defer-stream-wg/discussions/48

If the client cannot support a multipart response, and the accept header reflects this — then dont send it. For example using an older version of graphiql that doesn't support defer stream, but the server its targeting does.

File: spec/GraphQLOverHTTP.md (hunk 1 above)
Comment (michaelstaib) [reply spec/GraphQLOverHTTP.md:430]: The issue is that if you have a query that has deferred parts in it you have to respond with `MultiPart/Mixed` or an event stream.

**Comment by michaelstaib** [comment]
@maraisr that is exactly how I feel :D I would like it if we made it strict, so that if you send in the accept header that is what the server abides to. This would give predictability.

**Comment by benjie** [comment]
Yeah, this is just meant to be a fix; the major change in behaviour needs more discussion 👍

File: spec/GraphQLOverHTTP.md (hunk 1 above)
Comment (benjie) [reply spec/GraphQLOverHTTP.md:430]: I think there’s enough context in the rest of the spec to infer the suggestion.

**Comment by Shane32** [comment]
My thoughts:
- There are multiple methods to stream responses, such as the graphql-ws protocol or multipart/mixed
- I personally would expect that by default every GraphQL server responds in a basic (json) format, and does not prefer one streaming format over another -- although currently the spec states this:

> If a client does not supply the Accept header then the server may respond with an error, or with any content type it chooses

- Existing text would force a compliant server that receives a `@defer` or `@stream` request to return an `application/json` format.  Seems that it should be allowed to return any format, such as would be the case if there were no `Accept` header.
- HOWEVER, the written text only applies when the client specifically requests an _unsupported_ format.  As such, there should be no expectation that the client can interpret the response at all.  (Otherwise it would have specified the format in the `Accept` header.)
- In the sample at the top of this PR, it is quite unlikely that the client is going to be able to parse a multipart response.  If it were able to do so, it would/should have been specified in the request.  Responding with 406 would seem appropriate.

In conclusion, I agree with @benjie 's latest suggestion, that the spec should state that the server SHOULD return 406 (when the client requests an unsupported format), but MAY return any other format - both to mimic the HTTP spec and match the behavior when no `Accept` header is specified.

**Comment by benjie** [comment]
I've updated the text to reflect this.

**Review by JoviDeCroock — APPROVED** [review]

**Review by ghmcadams — APPROVED** [review]
